File: src/SimpleApp/Simple.Interfaces/Enums/FieldTypeEnum.cs
using System.ComponentModel;
using Simple.Utils.Attributes;

namespace Simple.AdminApplication.TemplateMng.Enums
{
    [GenerateTypeScript]
    public static class FieldTypeEnum
    {
        /// <summary>
        /// 文本
        /// </summary>
        [Description("文本")]
        public static readonly int Text = 1;

        /// <summary>
        /// 数字
        /// </summary>
        [Description("数字")]
        public static readonly int Number = 2;
        
        /// <summary>
        /// 日期
        /// </summary>
        [Description("日期")]
        public static readonly int Date = 3;
     
        /// <summary>
        /// 布尔
        /// </summary>
        [Description("布尔")]
        public static readonly int Bool = 4;
     
        /// <summary>
        /// 枚举
        /// </summary>
        [Description("枚举")]
        public static readonly int Enum = 5;
     
        /// <summary>
        /// 图片
        /// </summary>
        [Description("图片")]
        public static readonly int Image = 6;
    }
}

File: vue-vben-admin/apps/web-naive/src/api/enum/FieldTypeEnum.ts
export enum FieldTypeEnum {
  Text = 1,    // 文本
  Number = 2,    // 数字
  Date = 3,    // 日期
  Bool = 4,    // 布尔
  Enum = 5,    // 枚举
  Image = 6,    // 图片
}

export const FieldTypeEnumOption = [
  { label: '文本', value: 1 },
  { label: '数字', value: 2 },
  { label: '日期', value: 3 },
  { label: '布尔', value: 4 },
  { label: '枚举', value: 5 },
  { label: '图片', value: 6 },
];

export function getFieldTypeLabel(value: number): string | undefined {
  const item = FieldTypeEnumOption.find(x => x.value === value);
  return item?.label;
}
